src/language-js/print/call-arguments.js

```diff
@@ -95,40 +90,42 @@ function printCallArguments(path, options, print) {
     return allArgsBrokenOut();
   }
 
-  const shouldGroupFirst = shouldGroupFirstArg(args);
-  const shouldGroupLast = shouldGroupLastArg(args, options);
-  if (shouldGroupFirst || shouldGroupLast) {
-    if (
-      shouldGroupFirst
-        ? printedArguments.slice(1).some(willBreak)
-        : printedArguments.slice(0, -1).some(willBreak)
-    ) {
+  if (shouldGroupFirstArg(args)) {
+    const tailArgs = printedArguments.slice(1);
+    if (tailArgs.some(willBreak)) {
       return allArgsBrokenOut();
     }
+    let firstArg;
+    const firstArgSelector = isDynamicImport ? "source" : ["arguments", 0];
+    try {
+      firstArg = [print(firstArgSelector, { expandFirstArg: true }), ",", line];
+    } catch (caught) {
+      if (caught instanceof ArgExpansionBailout) {
+        return allArgsBrokenOut();
+      }
+      /* istanbul ignore next */
+      throw caught;
+    }
 
-    // We want to print the last argument with a special flag
-    let printedExpanded = [];
+    return conditionalGroup([
+      ["(", firstArg, ...tailArgs, ")"],
+      allArgsBrokenOut(),
+    ]);
+  }
 
+  if (shouldGroupLastArg(args, options)) {
+    const headArgs = printedArguments.slice(0, -1);
+    if (headArgs.some(willBreak)) {
+      return allArgsBrokenOut();
+    }
+    let lastArg;
+    const lastArgSelector = isDynamicImport
```

Comment:
Same here, maybe `getCallArgumentSelector(index)`, allow index to be nagative value (-1).